src/main/scala/Cqrs/DbAdapters/InMemoryDb.scala

```diff
@@ -0,0 +1,147 @@
+package Cqrs.DbAdapters
+
+import Cqrs.Aggregate
+
+import cats.data.Xor
+import cats.Monad
+import cats.arrow.NaturalTransformation
+import cats._
+import cats.free.Free
+import cats.state._
+import cats.free.Free.{ pure, liftF }
+
+import cats.std.all._
+
+import scala.collection.immutable.TreeMap
+
+
+object InMemoryDb {
+
+  trait EventSerialisation[E] {
+    def serialise(d: E): String
+    def unserialise(s: String): E
```

Comment:
Encode/decode might be better names here. Decoding might fail, so use `Option` or `Either` here I guess